src/scene/procedural.js

```diff
@@ -1041,6 +1042,11 @@ function getShapePrimitive(device, type) {
                 area = { x: Math.PI, y: Math.PI, z: Math.PI, uv: 1 };
                 break;
 
+            case 'torus':
+                mesh = createTorus(device, { tubeRadius: 0.2, ringRadius: 0.3 });
+                area = { x: Math.PI * 0.5 * 0.5 - Math.PI * 0.1 * 0.1, y: 0.4, z: 0.4, uv: 1 };
+                break;
+
```

Comment:
`getShapePrimitive` is only called from the `render` and `model` components. But those components don't list `torus` as a valid shape for `type`. Is your intention here to add support for it to those components?

Reply:
oh yes you're right the docs need updating.